### PR TITLE
Add PodDisruptionBudget to scheduler cache.

### PR DESCRIPTION
**Pull request by bsalamat**
**What this PR does / why we need it**:
This is the first step to add support for PodDisruptionBudget during preemption. This PR adds PDB to scheduler cache.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**: None

**Release note**:

```release-note
Add PodDisruptionBudget to scheduler cache.
```

ref/ #53913

### CODE REVIEW

**Comment by bsalamat** [comment]
@k82cn If you'd like to work on adding support for PDB to preemption, you may want to send a follow-up PR that checks PDB during preemption and node scoring.

**Comment by k82cn** [comment]
sure, I'll handle PDB part for preemption.

File: plugin/pkg/scheduler/factory/factory.go [hunk 1]
```diff
@@ -605,6 +621,56 @@ func (c *ConfigFactory) deleteNodeFromCache(obj interface{}) {
 	}
 }
 
+func (c *ConfigFactory) addPDBToCache(obj interface{}) {
```
Comment (resouer) [inline plugin/pkg/scheduler/factory/factory.go]: I think it's `configFactory`. Seems you need a rebase of latest change?

Comment (bsalamat) [reply plugin/pkg/scheduler/factory/factory.go]: Yes. Just fixed it.

**Comment by bsalamat** [comment]
@k82cn @resouer I fixed the RBAC rules which were causing tests to fail. PTAL.

**Comment by k82cn** [comment]
LGTM :).

**Comment by bsalamat** [comment]
@k82cn I had to [fix scheduler_perf/util.go](https://github.com/kubernetes/kubernetes/pull/53914/files#diff-8fc16044937fc03c39b585a657b20150) and add the new parameter for scheduler initialization. Could you please take another look?

**Comment by bsalamat** [comment]
@liggitt for approval of the changes in `auth`. Thanks!

**Comment by liggitt** [comment]
/approve
policy changes LGTM

**Pull request merged**